### PR TITLE
Replication: return results from simple queries + add more simple queries

### PR DESCRIPTION
**Pull request by greg-rychlewski**
This PR does the following:
- adds the ability to decode the rows returned by the simple queries executed on a replication connection
- returns the results from the existing simple queries, such as `create_slot` and `drop_slot` 
- adds the new commands `show`, `identify_server`, `timeline_history` and returns their results

### CODE REVIEW

**Comment by greg-rychlewski** [comment]
Converting to draft. I didn't realize the values from simple queries are all text. Need to change some things.

**Comment by greg-rychlewski** [comment]
All fixed.

File: lib/postgrex/replication.ex [hunk 1]
```diff
@@ -359,6 +370,77 @@ defmodule Postgrex.Replication do
...
+      Defaults to `5000`.
+
+  To better understand the meaning of this command,
+  [see PostgreSQL replication docs](https://www.postgresql.org/docs/14/protocol-replication.html).
+  """
+  @spec show(server, String.t(), Keyword.t()) ::
+          {:ok, Postgrex.Result.t()}
+          | {:error, Postgrex.Error.t()}
+          | {:error, :replication_started}
+  def show(pid, name, opts \\ []) do
```
Comment (josevalim) [inline lib/postgrex/replication.ex:391]: Should we check `name` is a string?

File: lib/postgrex/replication.ex [hunk 2]
```diff
@@ -359,6 +370,77 @@ defmodule Postgrex.Replication do
...
+      Defaults to `5000`.
+
+  To better understand the meaning of this command,
+  [see PostgreSQL replication docs](https://www.postgresql.org/docs/14/protocol-replication.html).
+  """
+  @spec timeline_history(server, String.t(), Keyword.t()) ::
+          {:ok, Postgrex.Result.t()}
+          | {:error, Postgrex.Error.t()}
+          | {:error, :replication_started}
+  def timeline_history(pid, timeline_id, opts \\ []) do
```
Comment (josevalim) [inline lib/postgrex/replication.ex:438]: Same here, check that timeline_id is a string?

File: test/replication_test.exs [hunk 3]
```diff
@@ -122,9 +130,23 @@ defmodule ReplicationTest do
...
+    {:ok, %Postgrex.Result{} = result} = PR.identify_system(context.repl)
+    assert result.num_rows == 1
+    refute is_nil(result.columns)
+    refute is_nil(result.rows)
+  end
+
+  test "show returns values", context do
+    {:ok, %Postgrex.Result{} = result} = PR.show(context.repl, "SERVER_VERSION")
+    assert result.num_rows == 1
+    refute is_nil(result.columns)
```
Comment (josevalim) [inline test/replication_test.exs:143]: Can  we assert on the column name? It should be static, right? For the rows, we can probably do `assert [[_]] = result.rows`?

File: test/replication_test.exs [hunk 4]
```diff
@@ -122,9 +130,23 @@ defmodule ReplicationTest do
     assert Exception.message(error) =~ "replication slot \"postgrex_example\" is active for PID"
   end
 
+  test "identify system returns values", context do
+    {:ok, %Postgrex.Result{} = result} = PR.identify_system(context.repl)
+    assert result.num_rows == 1
+    refute is_nil(result.columns)
```
Comment (josevalim) [inline test/replication_test.exs:136]: Can  we assert on the column name? It should be static, right? For the rows, we can probably do `assert [[_]] = result.rows`?

**Comment by josevalim** [comment]
Looks great! 🎉  I have added only some minor comments.

**Comment by greg-rychlewski** [comment]
Thanks! I double checked that the column names are consistent across versions for the output of `create_slot`, 'show` and `identify_system` and they are.

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**